### DISCUSSION TITLE
Word-wrapped text in form field not displaying in multilines

### DISCUSSION THREAD

**Discussion by zhiwei-arbo** [pikepdf/pikepdf#479] [Q&A]
Hi there, thanks for the awesome library.

I have a PDF object that looks like this:
```
<pikepdf.Dictionary(Type="/Annot")({
  "/AP": {
    "/N": pikepdf.Stream(owner=<...>, data=<...>, {
        "/BBox": [ 0, 0, 150, Decimal('28.2524') ],
        "/Filter": "/FlateDecode",
        "/Length": 117,
        "/Matrix": [ 1, 0, 0, 1, 0, 0 ],
        "/Resources": {
          "/Font": {
            "/F1": {
              "/BaseFont": "/Helvetica",
              "/Encoding": "/WinAnsiEncoding",
              "/Subtype": "/Type1",
              "/Type": "/Font"
            }
          }
        },
        "/Subtype": "/Form",
        "/Type": "/XObject"
      })
  },
  "/BS": {
    "/W": 0
  },
  "/DA": "/F1 8 Tf 0 0 0 rg ",
  "/DV": "",
  "/F": 4,
  "/FT": "/Tx",
  "/Ff": 8392704,
  "/H": "/N",
  "/MK": {

  },
  "/P": <Pdf.pages.from_objgen(205,0)>,
  "/Rect": [ Decimal('258.144'), Decimal('286.183'), Decimal('408.144'), Decimal('314.435') ],
  "/Subtype": "/Widget",
  "/T": "Some Field",
  "/Type": "/Annot",
  "/V": "This is a very long text and should render over multiple lines"
```

What ends up happening is it renders like this:
![image](https://github.com/pikepdf/pikepdf/assets/104612633/d6e2d2d5-e4aa-4740-95dd-f72ed71c1558)

Although if i click on the field in my PDF editor, it re-renders into multiple lines:
![image](https://github.com/pikepdf/pikepdf/assets/104612633/49a694f3-2af3-41dd-ab88-dcbea3d6752f)

Is there a way to make it render over multiple lines in the first place?

The only difference I see between the single line one vs the multi-line variant (I clicked on the form field with my PDF editor, then saved it into another file), is that the `/Length` attribute is much higher in the multi-line variant. However it doesn't seem to be a modifiable attribute

**Comment by jbarlow83** [comment]
The /Length attribute is associated with some compressed data. If you were allowed to change it the PDF would be corrupted. It has no bearing on the length of the text field.

I don't know if or how you can control this with any particular annotation attribute. It's very likely this is behavior varies from PDF editor to editor, and you'll find it difficult to influence. Have a look at the PDF 2.0 or 1.7 reference manual to see if information is available there.

**Reply by zhiwei-arbo** [reply]
any tips on where to start looking in the manual? I did take a look but saw nothing relevant in the section on string objects

**Reply by zhiwei-arbo** [reply]
From the 'Variable Text' section:
![image](https://github.com/pikepdf/pikepdf/assets/104612633/1653d655-6f5b-4ac2-9eec-e64765e1a468)


So it seems the appearance is generated by the reader in this case, and it might not be something i can control unfortunately?

**Reply by jbarlow83** [reply]
It's quite possible.

I often tell people that PDF viewers (especially interactive features like forms) are like dealing with web browsers in the early/mid 2000s when we had to deal with IE6-8... except much worse.
